src/neo/Consensus/ConsensusService.cs

```diff
@@ -185,8 +187,41 @@ private void InitializeConsensus(byte viewNumber)
             {
                 ChangeTimer(TimeSpan.FromMilliseconds(Blockchain.MillisecondsPerBlock << (viewNumber + 1)));
             }
+
+            // Try to speed up consensus if cached future payloads exists
+            if (context.HasFuturePayloads())
+            {
+                TryToConsumeFuturePayload(context.FuturePreparationPayloads);
+                if (context.HasFuturePayloads())
+                    TryToConsumeFuturePayload(context.FutureCommitPayloads);
+                if (context.HasFuturePayloads())
+                    TryToConsumeFuturePayload(context.FutureChangeViewPayloads);
+                if (context.HasFuturePayloads())
+                    TryToConsumeFuturePayload(context.FutureRecoveryPayloads);
```

Comment:
These recovery payloads are not requested by us, so we can ignore them.

Reply:
I do not agree, these are free information, why would we discard them? They can contain payloads that we may need to still receive.